Index: squid/src/client_side.c
diff -c squid/src/client_side.c:1.561.2.56 squid/src/client_side.c:1.561.2.57
*** squid/src/client_side.c:1.561.2.56	Sat Jul 17 10:01:50 2004
--- squid/src/client_side.c	Sat Jul 17 10:12:16 2004
***************
*** 1673,1679 ****
  	/*
  	 * plain ol' cache hit
  	 */
! 	if (e->mem_status == IN_MEMORY)
  	    http->log_type = LOG_TCP_MEM_HIT;
  	else if (Config.onoff.offline)
  	    http->log_type = LOG_TCP_OFFLINE_HIT;
--- 1673,1681 ----
  	/*
  	 * plain ol' cache hit
  	 */
! 	if (e->store_status != STORE_OK)
! 	    http->log_type = LOG_TCP_MISS;
! 	else if (e->mem_status == IN_MEMORY)
  	    http->log_type = LOG_TCP_MEM_HIT;
  	else if (Config.onoff.offline)
  	    http->log_type = LOG_TCP_OFFLINE_HIT;
