diff -urN squid-2.5.STABLE7.orig/src/defines.h squid-2.5.STABLE7/src/defines.h
--- squid-2.5.STABLE7.orig/src/defines.h	2002-08-08 23:17:39 +0300
+++ squid-2.5.STABLE7/src/defines.h	2004-10-12 23:54:32 +0300
@@ -219,7 +219,7 @@
 
 /* were to look for errors if config path fails */
 #ifndef DEFAULT_SQUID_ERROR_DIR
-#define DEFAULT_SQUID_ERROR_DIR "/usr/local/squid/etc/errors"
+#define DEFAULT_SQUID_ERROR_DIR "/usr/share/squid/errors/English"
 #endif
 
 /* gb_type operations */

diff -urN squid-2.6.STABLE5.orig/src/cf.data.pre squid-2.6.STABLE5/src/cf.data.pre
--- squid-2.6.STABLE5.orig/src/cf.data.pre	2004-10-08 20:41:10 +0300
+++ squid-2.6.STABLE5/src/cf.data.pre	2004-10-12 23:54:00 +0300
@@ -2802,6 +2802,7 @@
 acl localhost src 127.0.0.1/255.255.255.255
 acl to_localhost dst 127.0.0.0/8
 acl SSL_ports port 443
+acl Jabber_ports port 5222 5223
 acl Safe_ports port 80		# http
 acl Safe_ports port 21		# ftp
 acl Safe_ports port 443		# https
@@ -2934,7 +2935,7 @@
 # Deny requests to unknown ports
 http_access deny !Safe_ports
 # Deny CONNECT to other than SSL ports
-http_access deny CONNECT !SSL_ports
+http_access deny CONNECT !SSL_ports !Jabber_ports
 #
 # We strongly recommend the following be uncommented to protect innocent
 # web applications running on the proxy server who think the only
@@ -2950,6 +2951,7 @@
 #http_access allow our_networks
 
 # And finally deny all other access to this proxy
+http_access allow localhost
 http_access deny all
 NOCOMMENT_END
 DOC_END
@@ -3009,7 +3011,7 @@
 
 NOCOMMENT_START
 #Allow ICP queries from everyone
-icp_access allow all
+#icp_access allow all
 NOCOMMENT_END
 DOC_END
 
@@ -3280,7 +3282,7 @@
 
 NAME: cache_effective_user
 TYPE: string
-DEFAULT: nobody
+DEFAULT: squid
 LOC: Config.effectiveUser
 DOC_START
 	If you start Squid as root, it will change its effective/real
@@ -3471,7 +3473,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
