Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.93 squid/src/cf.data.pre:1.245.2.94
*** squid/src/cf.data.pre:1.245.2.93	Sun May  8 19:48:09 2005
--- squid/src/cf.data.pre	Tue May 10 17:08:40 2005
***************
*** 3080,3087 ****
  	Usage: always_direct allow|deny [!]aclname ...
  
  	Here you can use ACL elements to specify requests which should
! 	ALWAYS be forwarded directly to origin servers.  For example,
! 	to always directly forward requests for local servers use
  	something like:
  
  		acl local-servers dstdomain my.domain.net
--- 3080,3088 ----
  	Usage: always_direct allow|deny [!]aclname ...
  
  	Here you can use ACL elements to specify requests which should
! 	ALWAYS be forwarded by Squid to the origin servers without using
! 	any peers.  For example, to always directly forward requests for
! 	local servers ignoring any parents or siblings you may have use
  	something like:
  
  		acl local-servers dstdomain my.domain.net
***************
*** 3103,3108 ****
--- 3104,3118 ----
  		always_direct deny local-external
  		always_direct allow local-servers
  
+ 	NOTE: If your goal is to make the client forward the request
+ 	directly to the origin server bypassing Squid then this needs
+ 	to be done in the client configuration. Squid configuration
+ 	can only tell Squid how Squid should fetch the object.
+ 
+ 	NOTE: This directive is not related to caching. The replies
+ 	is cached as usual even if you use always_direct. To not cache
+ 	the replies see no_cache.
+ 
  	This option replaces some v1.1 options such as local_domain
  	and local_ip.
  DOC_END
