Index: squid/configure.in
diff -c squid/configure.in:1.251.2.59 squid/configure.in:1.251.2.60
*** squid/configure.in:1.251.2.59	Thu Jul  8 17:31:56 2004
--- squid/configure.in	Mon Aug  9 07:54:22 2004
***************
*** 1925,1930 ****
--- 1925,1931 ----
  	drand48 \
  	tempnam \
  	strerror \
+ 	initgroups
  )
  
  dnl Not cached since people are likely to tune this
Index: squid/include/initgroups.h
diff -c /dev/null squid/include/initgroups.h:1.1.2.1
*** /dev/null	Mon Aug  9 07:59:11 2004
--- squid/include/initgroups.h	Mon Aug  9 07:54:23 2004
***************
*** 0 ****
--- 1,17 ----
+ /*
+  * $Id$
+  */
+ #ifndef SQUID_INITGROUPS_H
+ #define SQUID_INITGROUPS_H
+ 
+ /* if you have configure you can use this */
+ #if defined(HAVE_CONFIG_H)
+ #include "config.h"
+ #endif
+ 
+ #if HAVE_SYS_TYPES_H
+ #include <sys/types.h>
+ #endif
+ 
+ extern int initgroups(const char *user, gid_t group);
+ #endif /* SQUID_INITGROPS_H */
Index: squid/lib/initgroups.c
diff -c /dev/null squid/lib/initgroups.c:1.1.2.1
*** /dev/null	Mon Aug  9 07:59:11 2004
--- squid/lib/initgroups.c	Mon Aug  9 07:54:23 2004
***************
*** 0 ****
--- 1,54 ----
+ #include "config.h"
+ 
+ #if HAVE_GRP_H
+ #include <grp.h>
+ #endif
+ #if HAVE_SYS_TYPES_H
+ #include <sys/types.h>
+ #endif
+ #if HAVE_UNISTD_H
+ #include <unistd.h>
+ #endif
+ #if HAVE_STRING_H
+ #include <string.h>
+ #endif
+ #if HAVE_STRINGS_H
+ #include <strings.h>
+ #endif
+ #if HAVE_LIMITS_H
+ #include <limits.h>
+ #endif
+ 
+ int initgroups(const char *name, gid_t basegid)
+ {
+ #ifdef HAVE_SETGROUPS
+ #ifndef NGROUPS_MAX
+ #define NGROUPS_MAX 16
+ #endif
+     gid_t groups[NGROUPS_MAX];
+     struct group *g;
+     int index = 0;
+ 
+     setgrent();
+ 
+     groups[index++] = basegid;
+ 
+     while (index < NGROUPS_MAX && ((g = getgrent()) != NULL)) {
+ 	if (g->gr_gid != basegid) {
+ 	    char **names;
+ 
+ 	    for (names = g->gr_mem; *names != NULL; ++names) {
+ 		if (!strcmp(*names, name))
+ 		    groups[index++] = g->gr_gid;
+ 	    }
+ 	}
+     }
+ 
+     endgrent();
+ 
+     return setgroups(index, groups);
+ #else
+     return 0;
+ #endif /* def HAVE_SETGROUPS */
+ }
+ 
Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.69 squid/src/cf.data.pre:1.245.2.70
*** squid/src/cf.data.pre:1.245.2.69	Wed Aug  4 10:13:26 2004
--- squid/src/cf.data.pre	Mon Aug  9 07:54:23 2004
***************
*** 2400,2426 ****
  TYPE: string
  DEFAULT: squid
  LOC: Config.effectiveUser
! DOC_NONE
  
  NAME: cache_effective_group
  TYPE: string
  DEFAULT: squid
  LOC: Config.effectiveGroup
  DOC_START
! 
! 	If you start Squid as root, it will change its effective/real
! 	UID/GID to the UID/GID specified below.  The default is to
! 	change to UID to nobody.  If you define cache_effective_user,
! 	but not cache_effective_group, Squid sets the GID the
! 	effective user's default group ID (taken from the password
! 	file).
! 
! 	If Squid is not started as root, the cache_effective_user
! 	value is ignored and the GID value is unchanged by default.
! 	However, you can make Squid change its GID to another group
! 	that the process owner is a member of.  Note that if Squid
! 	is not started as root then you cannot set http_port to a
! 	value lower than 1024.
  DOC_END
  
  
--- 2400,2428 ----
  TYPE: string
  DEFAULT: squid
  LOC: Config.effectiveUser
! DOC_START
! 	If you start Squid as root, it will change its effective/real
! 	UID/GID to the user specified below.  The default is to change
! 	to UID to nobody.  If you define cache_effective_user, but not
! 	cache_effective_group, Squid sets the GID to the effective
! 	user's default group ID (taken from the password file) and
! 	supplementary group list from the from groups membership of
! 	cache_effective_user.
! DOC_END
! 
  
  NAME: cache_effective_group
  TYPE: string
  DEFAULT: squid
  LOC: Config.effectiveGroup
  DOC_START
! 	If you want Squid to run with a specific GID regardless of
! 	the group memberships of the effective user then set this
! 	to the group (or GID) you want Squid to run as. When set
! 	all other group privileges of the effective user is ignored
! 	and only this GID is effective. If Squid is not started as
! 	root the user starting Squid must be member of the specified
! 	group.
  DOC_END
  
  
Index: squid/src/squid.h
diff -c squid/src/squid.h:1.216.2.6 squid/src/squid.h:1.216.2.7
*** squid/src/squid.h:1.216.2.6	Sun Mar  9 11:41:27 2003
--- squid/src/squid.h	Mon Aug  9 07:54:23 2004
***************
*** 420,425 ****
--- 420,429 ----
  #include "snprintf.h"
  #endif
  
+ #if !HAVE_INITGROUPS
+ #include "initgroups.h"
+ #endif
+ 
  #define XMIN(x,y) ((x)<(y)? (x) : (y))
  #define XMAX(a,b) ((a)>(b)? (a) : (b))
  
Index: squid/src/tools.c
diff -c squid/src/tools.c:1.213.2.8 squid/src/tools.c:1.213.2.9
*** squid/src/tools.c:1.213.2.8	Tue Jun 24 14:52:26 2003
--- squid/src/tools.c	Mon Aug  9 07:54:23 2004
***************
*** 516,533 ****
  leave_suid(void)
  {
      debug(21, 3) ("leave_suid: PID %d called\n", (int) getpid());
      if (geteuid() != 0)
  	return;
      /* Started as a root, check suid option */
      if (Config.effectiveUser == NULL)
  	return;
- #if HAVE_SETGROUPS
-     setgroups(1, &Config2.effectiveGroupID);
- #endif
-     if (setgid(Config2.effectiveGroupID) < 0)
- 	debug(50, 0) ("ALERT: setgid: %s\n", xstrerror());
      debug(21, 3) ("leave_suid: PID %d giving up root, becoming '%s'\n",
  	(int) getpid(), Config.effectiveUser);
  #if HAVE_SETRESUID
      if (setresuid(Config2.effectiveUserID, Config2.effectiveUserID, 0) < 0)
  	debug(50, 0) ("ALERT: setresuid: %s\n", xstrerror());
--- 516,544 ----
  leave_suid(void)
  {
      debug(21, 3) ("leave_suid: PID %d called\n", (int) getpid());
+     if (Config.effectiveGroup) {
+ #if HAVE_SETGROUPS
+ 	setgroups(1, &Config2.effectiveGroupID);
+ #endif
+ 	if (setgid(Config2.effectiveGroupID) < 0)
+ 	    debug(50, 0) ("ALERT: setgid: %s\n", xstrerror());
+     }
      if (geteuid() != 0)
  	return;
      /* Started as a root, check suid option */
      if (Config.effectiveUser == NULL)
  	return;
      debug(21, 3) ("leave_suid: PID %d giving up root, becoming '%s'\n",
  	(int) getpid(), Config.effectiveUser);
+     if (!Config.effectiveGroup) {
+ 	if (setgid(Config2.effectiveGroupID) < 0)
+ 	    debug(50, 0) ("ALERT: setgid: %s\n", xstrerror());
+ 	if (initgroups(Config.effectiveUser, Config2.effectiveGroupID) < 0) {
+ 	    debug(50, 0) ("ALERT: initgroups: unable to set groups for User %s "
+ 		"and Group %u", Config.effectiveUser,
+ 		(unsigned) Config2.effectiveGroupID);
+ 	}
+     }
  #if HAVE_SETRESUID
      if (setresuid(Config2.effectiveUserID, Config2.effectiveUserID, 0) < 0)
  	debug(50, 0) ("ALERT: setresuid: %s\n", xstrerror());
Index: squid/include/autoconf.h.in
diff -c squid/include/autoconf.h.in:1.109.2.9 squid/include/autoconf.h.in:1.109.2.10
*** squid/include/autoconf.h.in:1.109.2.9	Tue Jun  8 05:37:22 2004
--- squid/include/autoconf.h.in	Mon Aug  9 07:55:49 2004
***************
*** 470,475 ****
--- 470,478 ----
  /* Define if you have the getspnam function.  */
  #undef HAVE_GETSPNAM
  
+ /* Define if you have the initgroups function.  */
+ #undef HAVE_INITGROUPS
+ 
  /* Define if you have the lrand48 function.  */
  #undef HAVE_LRAND48
  
Index: squid/lib/Makefile.in
diff -c squid/lib/Makefile.in:1.57.2.10 squid/lib/Makefile.in:1.57.2.12
*** squid/lib/Makefile.in:1.57.2.10	Sat Jul 10 06:11:47 2004
--- squid/lib/Makefile.in	Mon Aug  9 07:57:38 2004
***************
*** 221,235 ****
  @AMDEP_TRUE@	$(DEPDIR)/dlmalloc.Po $(DEPDIR)/drand48.Po \
  @AMDEP_TRUE@	$(DEPDIR)/getfullhostname.Po $(DEPDIR)/hash.Po \
  @AMDEP_TRUE@	$(DEPDIR)/heap.Po $(DEPDIR)/html_quote.Po \
! @AMDEP_TRUE@	$(DEPDIR)/inet_ntoa.Po $(DEPDIR)/iso3307.Po \
! @AMDEP_TRUE@	$(DEPDIR)/md5.Po $(DEPDIR)/ntlmauth.Po \
! @AMDEP_TRUE@	$(DEPDIR)/radix.Po $(DEPDIR)/rfc1035.Po \
! @AMDEP_TRUE@	$(DEPDIR)/rfc1123.Po $(DEPDIR)/rfc1738.Po \
! @AMDEP_TRUE@	$(DEPDIR)/rfc2617.Po $(DEPDIR)/safe_inet_addr.Po \
! @AMDEP_TRUE@	$(DEPDIR)/snprintf.Po $(DEPDIR)/splay.Po \
! @AMDEP_TRUE@	$(DEPDIR)/strerror.Po $(DEPDIR)/stub_memaccount.Po \
! @AMDEP_TRUE@	$(DEPDIR)/tempnam.Po $(DEPDIR)/util.Po \
! @AMDEP_TRUE@	$(DEPDIR)/uudecode.Po
  COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
  	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
  CCLD = $(CC)
--- 221,235 ----
  @AMDEP_TRUE@	$(DEPDIR)/dlmalloc.Po $(DEPDIR)/drand48.Po \
  @AMDEP_TRUE@	$(DEPDIR)/getfullhostname.Po $(DEPDIR)/hash.Po \
  @AMDEP_TRUE@	$(DEPDIR)/heap.Po $(DEPDIR)/html_quote.Po \
! @AMDEP_TRUE@	$(DEPDIR)/inet_ntoa.Po $(DEPDIR)/initgroups.Po \
! @AMDEP_TRUE@	$(DEPDIR)/iso3307.Po $(DEPDIR)/md5.Po \
! @AMDEP_TRUE@	$(DEPDIR)/ntlmauth.Po $(DEPDIR)/radix.Po \
! @AMDEP_TRUE@	$(DEPDIR)/rfc1035.Po $(DEPDIR)/rfc1123.Po \
! @AMDEP_TRUE@	$(DEPDIR)/rfc1738.Po $(DEPDIR)/rfc2617.Po \
! @AMDEP_TRUE@	$(DEPDIR)/safe_inet_addr.Po $(DEPDIR)/snprintf.Po \
! @AMDEP_TRUE@	$(DEPDIR)/splay.Po $(DEPDIR)/strerror.Po \
! @AMDEP_TRUE@	$(DEPDIR)/stub_memaccount.Po $(DEPDIR)/tempnam.Po \
! @AMDEP_TRUE@	$(DEPDIR)/util.Po $(DEPDIR)/uudecode.Po
  COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
  	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
  CCLD = $(CC)
***************
*** 238,245 ****
  DIST_SOURCES = $(libdlmalloc_a_SOURCES) $(libmiscutil_a_SOURCES) \
  	$(EXTRA_libmiscutil_a_SOURCES) $(libntlmauth_a_SOURCES) \
  	$(libregex_a_SOURCES)
! DIST_COMMON = Makefile.am Makefile.in drand48.c inet_ntoa.c strerror.c \
! 	tempnam.c
  SOURCES = $(libdlmalloc_a_SOURCES) $(libmiscutil_a_SOURCES) $(EXTRA_libmiscutil_a_SOURCES) $(libntlmauth_a_SOURCES) $(libregex_a_SOURCES)
  
  all: all-am
--- 238,245 ----
  DIST_SOURCES = $(libdlmalloc_a_SOURCES) $(libmiscutil_a_SOURCES) \
  	$(EXTRA_libmiscutil_a_SOURCES) $(libntlmauth_a_SOURCES) \
  	$(libregex_a_SOURCES)
! DIST_COMMON = Makefile.am Makefile.in drand48.c inet_ntoa.c \
! 	initgroups.c strerror.c tempnam.c
  SOURCES = $(libdlmalloc_a_SOURCES) $(libmiscutil_a_SOURCES) $(EXTRA_libmiscutil_a_SOURCES) $(libntlmauth_a_SOURCES) $(libregex_a_SOURCES)
  
  all: all-am
***************
*** 292,297 ****
--- 292,298 ----
  @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/heap.Po@am__quote@
  @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/html_quote.Po@am__quote@
  @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/inet_ntoa.Po@am__quote@
+ @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/initgroups.Po@am__quote@
  @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/iso3307.Po@am__quote@
  @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/md5.Po@am__quote@
  @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/ntlmauth.Po@am__quote@
Index: squid/configure
diff -c squid/configure:1.248.2.60 squid/configure:1.248.2.61
*** squid/configure:1.248.2.60	Thu Jul  8 17:42:11 2004
--- squid/configure	Mon Aug  9 07:57:36 2004
***************
*** 7985,7990 ****
  	drand48 \
  	tempnam \
  	strerror \
  
  do
  echo $ac_n "checking for $ac_func""... $ac_c" 1>&6
--- 7985,7991 ----
  	drand48 \
  	tempnam \
  	strerror \
+ 	initgroups
  
  do
  echo $ac_n "checking for $ac_func""... $ac_c" 1>&6
