Index: squid/src/cache_cf.c
diff -c squid/src/cache_cf.c:1.396.2.23 squid/src/cache_cf.c:1.396.2.24
*** squid/src/cache_cf.c:1.396.2.23	Fri Mar 25 19:50:51 2005
--- squid/src/cache_cf.c	Wed Apr 20 15:52:26 2005
***************
*** 297,307 ****
      }
      fclose(fp);
      defaults_if_none();
!     configDoConfigure();
!     cachemgrRegister("config",
! 	"Current Squid Configuration",
! 	dump_config,
! 	1, 1);
      return err_count;
  }
  
--- 297,309 ----
      }
      fclose(fp);
      defaults_if_none();
!     if (opt_send_signal == -1) {
! 	configDoConfigure();
! 	cachemgrRegister("config",
! 	    "Current Squid Configuration",
! 	    dump_config,
! 	    1, 1);
!     }
      return err_count;
  }
  
Index: squid/src/globals.h
diff -c squid/src/globals.h:1.108.2.5 squid/src/globals.h:1.108.2.6
*** squid/src/globals.h:1.108.2.5	Fri Mar 25 19:50:53 2005
--- squid/src/globals.h	Wed Apr 20 15:52:26 2005
***************
*** 165,169 ****
--- 165,170 ----
  #if HAVE_SBRK
  extern void *sbrk_start;	/* 0 */
  #endif
+ extern int opt_send_signal;	/* -1 */
  
  #endif /* SQUID_GLOBALS_H */
Index: squid/src/main.c
diff -c squid/src/main.c:1.345.2.24 squid/src/main.c:1.345.2.25
*** squid/src/main.c:1.345.2.24	Wed Apr 20 09:54:55 2005
--- squid/src/main.c	Wed Apr 20 15:52:26 2005
***************
*** 38,44 ****
  /* for error reporting from xmalloc and friends */
  extern void (*failure_notify) (const char *);
  
- static int opt_send_signal = -1;
  static int opt_no_daemon = 0;
  static int opt_parse_cfg_only = 0;
  static char *opt_syslog_facility = NULL;
--- 38,43 ----
