---------------------
PatchSet 10319 
Date: 2006/05/12 15:58:50
Author: hno
Branch: SQUID_2_5
Tag: (none) 
Log:
Bug #1553: Memleak in HTCP client code

Members: 
	src/htcp.c:1.38.2.6->1.38.2.7 

Index: squid/src/htcp.c
===================================================================
RCS file: /cvsroot/squid/squid/src/htcp.c,v
retrieving revision 1.38.2.6
retrieving revision 1.38.2.7
diff -u -r1.38.2.6 -r1.38.2.7
--- squid/src/htcp.c	26 Mar 2005 02:50:53 -0000	1.38.2.6
+++ squid/src/htcp.c	12 May 2006 15:58:50 -0000	1.38.2.7
@@ -598,6 +598,9 @@
 	packerClean(&p);
     }
     pkt = htcpBuildPacket(&stuff, &pktlen);
+    safe_free(stuff.D.resp_hdrs);
+    safe_free(stuff.D.entity_hdrs);
+    safe_free(stuff.D.cache_hdrs);
     if (pkt == NULL) {
 	debug(31, 0) ("htcpTstReply: htcpBuildPacket() failed\n");
 	return;
