diff -urN squid/helpers/negotiate_auth/squid_kerb_auth/Makefile.in squid.new/helpers/negotiate_auth/squid_kerb_auth/Makefile.in
--- squid/helpers/negotiate_auth/squid_kerb_auth/Makefile.in	2010-08-24 05:43:34 +0000
+++ squid.new/helpers/negotiate_auth/squid_kerb_auth/Makefile.in	2010-08-31 14:52:11 +0000
@@ -62,13 +62,13 @@
 squid_kerb_auth_OBJECTS = $(am_squid_kerb_auth_OBJECTS)
 squid_kerb_auth_DEPENDENCIES =
 squid_kerb_auth_LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(squid_kerb_auth_LDFLAGS) $(LDFLAGS) -o $@
+	$(squid_kerb_auth_LDFLAGS) -o $@
 am_squid_kerb_auth_test_OBJECTS = squid_kerb_auth_test.$(OBJEXT) \
 	base64.$(OBJEXT)
 squid_kerb_auth_test_OBJECTS = $(am_squid_kerb_auth_test_OBJECTS)
 squid_kerb_auth_test_DEPENDENCIES =
 squid_kerb_auth_test_LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(squid_kerb_auth_test_LDFLAGS) $(LDFLAGS) -o $@
+	$(squid_kerb_auth_test_LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -282,10 +282,10 @@
 	-test -z "$(libexec_PROGRAMS)" || rm -f $(libexec_PROGRAMS)
 squid_kerb_auth$(EXEEXT): $(squid_kerb_auth_OBJECTS) $(squid_kerb_auth_DEPENDENCIES) 
 	@rm -f squid_kerb_auth$(EXEEXT)
-	$(squid_kerb_auth_LINK) $(squid_kerb_auth_OBJECTS) $(squid_kerb_auth_LDADD) $(LIBS)
+	$(squid_kerb_auth_LINK) $(squid_kerb_auth_OBJECTS) $(squid_kerb_auth_LDADD) $(LIBS) $(LDFLAGS)
 squid_kerb_auth_test$(EXEEXT): $(squid_kerb_auth_test_OBJECTS) $(squid_kerb_auth_test_DEPENDENCIES) 
 	@rm -f squid_kerb_auth_test$(EXEEXT)
-	$(squid_kerb_auth_test_LINK) $(squid_kerb_auth_test_OBJECTS) $(squid_kerb_auth_test_LDADD) $(LIBS)
+	$(squid_kerb_auth_test_LINK) $(squid_kerb_auth_test_OBJECTS) $(squid_kerb_auth_test_LDADD) $(LIBS) $(LDFLAGS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
