diff -Nur squid-2.5.STABLE3.orig/Makefile.am squid-2.5.STABLE3/Makefile.am
--- squid-2.5.STABLE3.orig/Makefile.am	2003-07-11 13:53:18 +0600
+++ squid-2.5.STABLE3/Makefile.am	2003-07-11 13:53:54 +0600
@@ -41,5 +41,5 @@
 	include/samba/README.txt
 
 install-pinger:
-	chown root $(DEFAULT_PINGER) 
-	chmod 4711 $(DEFAULT_PINGER)
+#	chown root $(DEFAULT_PINGER) 
+#	chmod 4711 $(DEFAULT_PINGER)
diff -Nur squid-2.5.STABLE3.orig/Makefile.in squid-2.5.STABLE3/Makefile.in
--- squid-2.5.STABLE3.orig/Makefile.in	2003-07-11 13:53:18 +0600
+++ squid-2.5.STABLE3/Makefile.in	2003-07-11 13:54:46 +0600
@@ -483,8 +483,8 @@
 	fi
 
 install-pinger:
-	chown root $(DEFAULT_PINGER) 
-	chmod 4711 $(DEFAULT_PINGER)
+#	chown root $(DEFAULT_PINGER) 
+#	chmod 4711 $(DEFAULT_PINGER)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Nur squid-2.5.STABLE3.orig/src/Makefile.in squid-2.5.STABLE3/src/Makefile.in
--- squid-2.5.STABLE3.orig/src/Makefile.in	2003-07-11 13:53:18 +0600
+++ squid-2.5.STABLE3/src/Makefile.in	2003-07-11 13:58:15 +0600
@@ -374,12 +374,12 @@
 DEFAULT_CONFIG_FILE = $(sysconfdir)/squid.conf
 DEFAULT_MIME_TABLE = $(sysconfdir)/mime.conf
 DEFAULT_DNSSERVER = $(libexecdir)/dnsserver$(EXEEXT)
-DEFAULT_LOG_PREFIX = $(localstatedir)/logs
+DEFAULT_LOG_PREFIX = $(localstatedir)/log/squid
 DEFAULT_CACHE_LOG = $(DEFAULT_LOG_PREFIX)/cache.log
 DEFAULT_ACCESS_LOG = $(DEFAULT_LOG_PREFIX)/access.log
 DEFAULT_STORE_LOG = $(DEFAULT_LOG_PREFIX)/store.log
-DEFAULT_PID_FILE = $(DEFAULT_LOG_PREFIX)/squid.pid
-DEFAULT_SWAP_DIR = $(localstatedir)/cache
+DEFAULT_PID_FILE = $(localstatedir)/run/squid.pid
+DEFAULT_SWAP_DIR = $(localstatedir)/spool/squid
 DEFAULT_PINGER = $(libexecdir)/pinger$(EXEEXT)
 DEFAULT_UNLINKD = $(libexecdir)/unlinkd$(EXEEXT)
 DEFAULT_DISKD = $(libexecdir)/diskd$(EXEEXT)
diff -Nur squid-2.5.STABLE3.orig/helpers/basic_auth/SASL/Makefile.in squid-2.5.STABLE3/helpers/basic_auth/SASL/Makefile.in
--- squid-2.5.STABLE3.orig/helpers/basic_auth/SASL/Makefile.in	2003-07-11 13:53:18 +0600
+++ squid-2.5.STABLE3/helpers/basic_auth/SASL/Makefile.in	2003-07-11 16:41:18 +0600
@@ -120,7 +120,7 @@
 install_sh = @install_sh@
 makesnmplib = @makesnmplib@
 
-INCLUDES = -I$(top_srcdir)/include
+INCLUDES = -I$(top_srcdir)/include -I$(includedir)/sasl
 
 libexec_PROGRAMS = sasl_auth
 LDADD = -L$(top_builddir)/lib -lmiscutil -lsasl $(XTRA_LIBS)
