Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.86 squid/src/cf.data.pre:1.245.2.87
*** squid/src/cf.data.pre:1.245.2.86	Fri Mar  4 17:45:10 2005
--- squid/src/cf.data.pre	Wed Mar  9 07:22:33 2005
***************
*** 3448,3457 ****
  LOC: Config.Delay
  DOC_START
  	This is used to determine which delay pool a request falls into.
! 	The first matched delay pool is always used, i.e., if a request falls
! 	into delay pool number one, no more delay are checked, otherwise the
! 	rest are checked in order of their delay pool number until they have
! 	all been checked.  For example, if you want some_big_clients in delay
  	pool 1 and lotsa_little_clients in delay pool 2:
  
  Example:
--- 3448,3460 ----
  LOC: Config.Delay
  DOC_START
  	This is used to determine which delay pool a request falls into.
! 
! 	delay_access is sorted per pool and the matching starts with pool 1,
!        	then pool 2, ..., and finally pool N. The first delay pool where the
!        	request is allowed is selected for the request. If it does not allow
! 	the request to any pool then the request is not delayed (default).
! 
!       	For example, if you want some_big_clients in delay
  	pool 1 and lotsa_little_clients in delay pool 2:
  
  Example:
