diff -urN squid/Makefile.in squid.new/Makefile.in
--- squid/Makefile.in	2010-08-24 05:42:46 +0000
+++ squid.new/Makefile.in	2010-08-31 15:05:38 +0000
@@ -764,8 +764,8 @@
 	fi
 
 install-pinger:
-	chown root $(DESTDIR)$(DEFAULT_PINGER)
-	chmod 4711 $(DESTDIR)$(DEFAULT_PINGER)
+	echo "$(DESTDIR)$(DEFAULT_PINGER) must be owned root"
+	echo "$(DESTDIR)$(DEFAULT_PINGER) must be chmod 4711"
 
 check: have-cppunit check-recursive
 
