Index: squid/src/fs/diskd/store_dir_diskd.c
diff -c squid/src/fs/diskd/store_dir_diskd.c:1.58.2.11 squid/src/fs/diskd/store_dir_diskd.c:1.58.2.12
*** squid/src/fs/diskd/store_dir_diskd.c:1.58.2.11	Sat Mar 26 16:27:11 2005
--- squid/src/fs/diskd/store_dir_diskd.c	Sun May  1 04:48:07 2005
***************
*** 452,469 ****
      storeAppendPrintf(sentry, "open_fail_queue_len: %d\n", diskd_stats.open_fail_queue_len);
      storeAppendPrintf(sentry, "block_queue_len: %d\n", diskd_stats.block_queue_len);
      diskd_stats.max_away = diskd_stats.max_shmuse = 0;
!     storeAppendPrintf(sentry, "\n             OPS SUCCESS    FAIL\n");
!     storeAppendPrintf(sentry, "%7s %7d %7d %7d\n",
  	"open", diskd_stats.open.ops, diskd_stats.open.success, diskd_stats.open.fail);
!     storeAppendPrintf(sentry, "%7s %7d %7d %7d\n",
  	"create", diskd_stats.create.ops, diskd_stats.create.success, diskd_stats.create.fail);
!     storeAppendPrintf(sentry, "%7s %7d %7d %7d\n",
  	"close", diskd_stats.close.ops, diskd_stats.close.success, diskd_stats.close.fail);
!     storeAppendPrintf(sentry, "%7s %7d %7d %7d\n",
  	"unlink", diskd_stats.unlink.ops, diskd_stats.unlink.success, diskd_stats.unlink.fail);
!     storeAppendPrintf(sentry, "%7s %7d %7d %7d\n",
  	"read", diskd_stats.read.ops, diskd_stats.read.success, diskd_stats.read.fail);
!     storeAppendPrintf(sentry, "%7s %7d %7d %7d\n",
  	"write", diskd_stats.write.ops, diskd_stats.write.success, diskd_stats.write.fail);
  }
  
--- 452,469 ----
      storeAppendPrintf(sentry, "open_fail_queue_len: %d\n", diskd_stats.open_fail_queue_len);
      storeAppendPrintf(sentry, "block_queue_len: %d\n", diskd_stats.block_queue_len);
      diskd_stats.max_away = diskd_stats.max_shmuse = 0;
!     storeAppendPrintf(sentry, "\n              OPS   SUCCESS    FAIL\n");
!     storeAppendPrintf(sentry, "%7s %9d %9d %7d\n",
  	"open", diskd_stats.open.ops, diskd_stats.open.success, diskd_stats.open.fail);
!     storeAppendPrintf(sentry, "%7s %9d %9d %7d\n",
  	"create", diskd_stats.create.ops, diskd_stats.create.success, diskd_stats.create.fail);
!     storeAppendPrintf(sentry, "%7s %9d %9d %7d\n",
  	"close", diskd_stats.close.ops, diskd_stats.close.success, diskd_stats.close.fail);
!     storeAppendPrintf(sentry, "%7s %9d %9d %7d\n",
  	"unlink", diskd_stats.unlink.ops, diskd_stats.unlink.success, diskd_stats.unlink.fail);
!     storeAppendPrintf(sentry, "%7s %9d %9d %7d\n",
  	"read", diskd_stats.read.ops, diskd_stats.read.success, diskd_stats.read.fail);
!     storeAppendPrintf(sentry, "%7s %9d %9d %7d\n",
  	"write", diskd_stats.write.ops, diskd_stats.write.success, diskd_stats.write.fail);
  }
  
