diff -urN squid-2.5.STABLE7.orig/src/defines.h squid-2.5.STABLE7/src/defines.h
--- squid-2.5.STABLE7.orig/src/defines.h	2002-08-08 23:17:39 +0300
+++ squid-2.5.STABLE7/src/defines.h	2004-10-12 23:54:32 +0300
@@ -219,7 +219,7 @@
 
 /* were to look for errors if config path fails */
 #ifndef DEFAULT_SQUID_ERROR_DIR
-#define DEFAULT_SQUID_ERROR_DIR "/usr/local/squid/etc/errors"
+#define DEFAULT_SQUID_ERROR_DIR "/usr/share/squid/errors/English"
 #endif
 
 /* gb_type operations */

diff -urN squid-2.6.STABLE5.orig/src/cf.data.pre squid-2.6.STABLE5/src/cf.data.pre
--- squid-2.6.STABLE5.orig/src/cf.data.pre	2004-10-08 20:41:10 +0300
+++ squid-2.6.STABLE5/src/cf.data.pre	2004-10-12 23:54:00 +0300
@@ -321,12 +321,13 @@
 NAME: htcp_port
 IFDEF: USE_HTCP
 TYPE: ushort
-DEFAULT: 4827
+DEFAULT: 0
 LOC: Config.Port.htcp
 DOC_START
 	The port number where Squid sends and receives HTCP queries to
-	and from neighbor caches.  Default is 4827.  To disable use
-	"0".
+	and from neighbor caches. A value of "0" disables HTCP support.
+	If you wish to use HTCP, set this to "4827" to use the normal
+	HTCP support.
 DOC_END
 
 
@@ -2801,7 +2808,10 @@
 acl manager proto cache_object
 acl localhost src 127.0.0.1/255.255.255.255
 acl to_localhost dst 127.0.0.0/8
-acl SSL_ports port 443
+acl SSL_ports port 443		# https
+acl SSL_ports port 563		# snews
+acl SSL_ports port 873		# rsync
+acl Jabber_ports port 5222 5223
 acl Safe_ports port 80		# http
 acl Safe_ports port 21		# ftp
 acl Safe_ports port 443		# https
@@ -2812,6 +2821,10 @@
 acl Safe_ports port 488		# gss-http
+acl Safe_ports port 563		# snews
 acl Safe_ports port 591		# filemaker
 acl Safe_ports port 777		# multiling http
+acl Safe_ports port 631		# cups
+acl Safe_ports port 873		# rsync
+acl Safe_ports port 901		# SWAT
 acl CONNECT method CONNECT
 NOCOMMENT_END
 DOC_END
@@ -2934,7 +2935,7 @@
 # Deny requests to unknown ports
 http_access deny !Safe_ports
 # Deny CONNECT to other than SSL ports
-http_access deny CONNECT !SSL_ports
+http_access deny CONNECT !SSL_ports !Jabber_ports
 #
 # We strongly recommend the following be uncommented to protect innocent
 # web applications running on the proxy server who think the only
@@ -2950,6 +2951,7 @@
 #http_access allow our_networks
 
 # And finally deny all other access to this proxy
+http_access allow localhost
 http_access deny all
 NOCOMMENT_END
 DOC_END
@@ -3009,7 +3011,7 @@
 
 NOCOMMENT_START
 #Allow ICP queries from everyone
-icp_access allow all
+#icp_access allow all
 NOCOMMENT_END
 DOC_END
 
@@ -2471,22 +2472,23 @@
 
 NAME: cache_effective_user
 TYPE: string
-DEFAULT: nobody
+DEFAULT: squid
 LOC: Config.effectiveUser
 DOC_START
 	If you start Squid as root, it will change its effective/real
 	UID/GID to the user specified below.  The default is to change
-	to UID to nobody.  If you define cache_effective_user, but not
+	to UID to "squid".  If you define cache_effective_user, but not
 	cache_effective_group, Squid sets the GID to the effective
 	user's default group ID (taken from the password file) and
 	supplementary group list from the from groups membership of
 	cache_effective_user.
+cache_effective_user squid
 DOC_END
 
 
 NAME: cache_effective_group
 TYPE: string
-DEFAULT: none
+DEFAULT: squid 
 LOC: Config.effectiveGroup
 DOC_START
 	If you want Squid to run with a specific GID regardless of
@@ -2496,6 +2498,7 @@
 	and only this GID is effective. If Squid is not started as
 	root the user starting Squid must be member of the specified
 	group.
+cache_effective_group squid
 DOC_END
 
 
@@ -3471,7 +3473,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
@@ -2739,6 +2742,8 @@
 	purposes, so -k rotate uses another signal.  It is best to get
 	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 	<pid>'.
+	
+logfile_rotate 0
 DOC_END
 
 
@@ -3437,6 +3438,10 @@
 	purposes, so -k rotate uses another signal.  It is best to get
 	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 	<pid>'.
+
+	ALT Linux is using the logrotate mechanism and therefore the
+	rotation is done externaly, which means a default of 0 is
+	required and therefore set. Modify /etc/logrotate.d/squid instead.
 DOC_END
 
 
@@ -3296,10 +3301,18 @@
 LOC: Config.errorDirectory
 DEFAULT: @DEFAULT_ERROR_DIR@
 DOC_START
-	If you wish to create your own versions of the default
-	(English) error files, either to customize them to suit your
-	language or company copy the template English files to another
+	Directory where the error files are read from.
+	/usr/lib/squid/errors contains sets of error files
+	in different languages. The default error directory
+	is /etc/squid/errors, which is a link to one of these
+	error sets.
+
+	If you wish to create your own versions of the error files,
+	either to customize them to suit your language or company,
+	copy the template English files to another
 	directory and point this tag at them.
+	
+error_directory @DEFAULT_ERROR_DIR@
 DOC_END
 
 NAME: maximum_single_addr_tries
@@ -3333,12 +3346,12 @@
 NAME: snmp_port
 TYPE: ushort
 LOC: Config.Port.snmp
-DEFAULT: 3401
+DEFAULT: 0
 IFDEF: SQUID_SNMP
 DOC_START
 	Squid can now serve statistics and status information via SNMP.
-	By default it listens to port 3401 on the machine. If you don't
-	wish to use SNMP, set this to "0".
+	A value of "0" disables SNMP support. If you wish to use SNMP,
+	set this to "3401" to use the normal SNMP support.
 DOC_END
 
 NAME: snmp_access
