diff --git a/squid/src/cf.data.pre b/squid/src/cf.data.pre
index 442447a..2b548ca 100644
--- a/squid/src/cf.data.pre
+++ b/squid/src/cf.data.pre
@@ -2722,9 +2722,13 @@ DOC_END
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
+	ALT Linux is using the logrotate mechanism and therefore the
+	rotation is done externaly, which means a default of 0 is
+	required and therefore set. Modify /etc/logrotate.d/squid instead.
+
 	Specifies the number of logfile rotations to make when you
 	type 'squid -k rotate'. The default is 10, which will rotate
 	with extensions 0 through 9. Setting logfile_rotate to 0 will
