Index: squid/src/store_swapout.c
diff -c squid/src/store_swapout.c:1.85.2.10 squid/src/store_swapout.c:1.85.2.11
*** squid/src/store_swapout.c:1.85.2.10	Sat Apr 30 06:40:28 2005
--- squid/src/store_swapout.c	Tue May 10 16:30:33 2005
***************
*** 113,119 ****
  	return 1;
  
      swapout_able = storeSwapOutAble(e);
!     if (!swapout_able && EBIT_TEST(e->flags, ENTRY_CACHABLE)) {
  	/* Stop writing to disk */
  	storeReleaseRequest(e);
  	if (e->mem_obj->swapout.sio != NULL)
--- 113,119 ----
  	return 1;
  
      swapout_able = storeSwapOutAble(e);
!     if (!swapout_able) {
  	/* Stop writing to disk */
  	storeReleaseRequest(e);
  	if (e->mem_obj->swapout.sio != NULL)
