Index: squid/configure.in
diff -c squid/configure.in:1.251.2.83 squid/configure.in:1.251.2.84
*** squid/configure.in:1.251.2.83	Fri Mar 18 18:10:27 2005
--- squid/configure.in	Fri Mar 18 18:31:35 2005
***************
*** 804,811 ****
  ])
  
  dnl Disable hostname checks
! enable_hostname_checks=1
! AC_ARG_ENABLE(hostname_checks,
  [  --disable-hostname-checks
                            Squid by default rejects any host names with
  			  odd characters in their name to conform with
--- 804,810 ----
  ])
  
  dnl Disable hostname checks
! AC_ARG_ENABLE(hostname-checks,
  [  --disable-hostname-checks
                            Squid by default rejects any host names with
  			  odd characters in their name to conform with
***************
*** 815,827 ****
  			  Squid does not reject such host names.. This
  			  may be required to participate in testbeds for
  			  international domain names.],
! [ if test "$enableval" = "no"; then
!     echo "Disabling hostname sanity checks"
!     enable_hostname_checks=0
!   fi
! ])
! if test "$enable_hostname_checks" = 1; then
    AC_DEFINE(CHECK_HOSTNAMES, 1, [Enable hostname sanity checks])
  fi
  
  dnl Enable underscore in hostnames
--- 814,826 ----
  			  Squid does not reject such host names.. This
  			  may be required to participate in testbeds for
  			  international domain names.],
! 	[],
! 	[enable_hostname_checks=yes])
! if test "$enable_hostname_checks" = "yes"; then
!   echo "Hostname sanity checks enabled"
    AC_DEFINE(CHECK_HOSTNAMES, 1, [Enable hostname sanity checks])
+ else
+   echo "Disabling hostname sanity checks"
  fi
  
  dnl Enable underscore in hostnames
Index: squid/configure
diff -c squid/configure:1.248.2.84 squid/configure:1.248.2.85
*** squid/configure:1.248.2.84	Fri Mar 18 18:10:52 2005
--- squid/configure	Fri Mar 18 18:31:54 2005
***************
*** 2687,2708 ****
  fi
  
  
! enable_hostname_checks=1
! # Check whether --enable-hostname_checks or --disable-hostname_checks was given.
  if test "${enable_hostname_checks+set}" = set; then
    enableval="$enable_hostname_checks"
!    if test "$enableval" = "no"; then
!     echo "Disabling hostname sanity checks"
!     enable_hostname_checks=0
!   fi
! 
  fi
  
! if test "$enable_hostname_checks" = 1; then
    cat >> confdefs.h <<\EOF
  #define CHECK_HOSTNAMES 1
  EOF
  
  fi
  
  # Check whether --enable-underscores or --disable-underscores was given.
--- 2687,2708 ----
  fi
  
  
! # Check whether --enable-hostname-checks or --disable-hostname-checks was given.
  if test "${enable_hostname_checks+set}" = set; then
    enableval="$enable_hostname_checks"
!   :
! else
!   enable_hostname_checks=yes
  fi
  
! if test "$enable_hostname_checks" = "yes"; then
!   echo "Hostname sanity checks enabled"
    cat >> confdefs.h <<\EOF
  #define CHECK_HOSTNAMES 1
  EOF
  
+ else
+   echo "Disabling hostname sanity checks"
  fi
  
  # Check whether --enable-underscores or --disable-underscores was given.
