Index: squid/src/disk.c
diff -c squid/src/disk.c:1.157.2.5 squid/src/disk.c:1.157.2.6
*** squid/src/disk.c:1.157.2.5	Sat Mar 26 10:36:01 2005
--- squid/src/disk.c	Sat Mar 26 16:27:10 2005
***************
*** 91,97 ****
  	read_callback(-1, F->read_data);
      }
      if (F->flags.write_daemon) {
! #if defined(_SQUID_MSWIN_) || defined(_SQUID_OS2_) || defined (_SQUID_CYGWIN_)
  	/*
  	 * on some operating systems, you can not delete or rename
  	 * open files, so we won't allow delayed close.
--- 91,97 ----
  	read_callback(-1, F->read_data);
      }
      if (F->flags.write_daemon) {
! #if defined(_SQUID_MSWIN_) || defined(_SQUID_OS2_) || defined(_SQUID_CYGWIN_)
  	/*
  	 * on some operating systems, you can not delete or rename
  	 * open files, so we won't allow delayed close.
Index: squid/src/tools.c
diff -c squid/src/tools.c:1.213.2.13 squid/src/tools.c:1.213.2.14
*** squid/src/tools.c:1.213.2.13	Fri Mar 25 19:50:54 2005
--- squid/src/tools.c	Sat Mar 26 16:27:10 2005
***************
*** 939,944 ****
--- 939,947 ----
  xrename(const char *from, const char *to)
  {
      debug(21, 2) ("xrename: renaming %s to %s\n", from, to);
+ #if defined(_SQUID_OS2_) || defined(_SQUID_CYGWIN_) || defined(_SQUID_MSWIN_)
+     remove(to);
+ #endif
      if (0 == rename(from, to))
  	return 0;
      debug(21, errno == ENOENT ? 2 : 1) ("xrename: Cannot rename %s to %s: %s\n",
Index: squid/src/fs/aufs/store_dir_aufs.c
diff -c squid/src/fs/aufs/store_dir_aufs.c:1.40.2.13 squid/src/fs/aufs/store_dir_aufs.c:1.40.2.14
*** squid/src/fs/aufs/store_dir_aufs.c:1.40.2.13	Sat Mar 26 15:29:25 2005
--- squid/src/fs/aufs/store_dir_aufs.c	Sat Mar 26 16:27:10 2005
***************
*** 1100,1111 ****
      char *new_path = xstrdup(storeAufsDirSwapLogFile(sd, ".new"));
      int fd;
      file_close(aioinfo->swaplog_fd);
- #if defined (_SQUID_OS2_) || defined (_SQUID_CYGWIN_)
-     if (unlink(swaplog_path) < 0) {
- 	debug(50, 0) ("%s: %s\n", swaplog_path, xstrerror());
- 	fatal("storeAufsDirCloseTmpSwapLog: unlink failed");
-     }
- #endif
      if (xrename(new_path, swaplog_path) < 0) {
  	fatal("storeAufsDirCloseTmpSwapLog: rename failed");
      }
--- 1100,1105 ----
***************
*** 1337,1345 ****
  #if defined(_SQUID_OS2_) || defined (_SQUID_CYGWIN_)
  	file_close(state->fd);
  	state->fd = -1;
- 	if (unlink(state->cur) < 0)
- 	    debug(50, 0) ("storeDirWriteCleanLogs: unlinkd failed: %s, %s\n",
- 		xstrerror(), state->cur);
  #endif
  	xrename(state->new, state->cur);
      }
--- 1331,1336 ----
Index: squid/src/fs/coss/store_dir_coss.c
diff -c squid/src/fs/coss/store_dir_coss.c:1.30.2.10 squid/src/fs/coss/store_dir_coss.c:1.30.2.11
*** squid/src/fs/coss/store_dir_coss.c:1.30.2.10	Fri Mar 25 19:50:55 2005
--- squid/src/fs/coss/store_dir_coss.c	Sat Mar 26 16:27:10 2005
***************
*** 411,422 ****
      char *new_path = xstrdup(storeCossDirSwapLogFile(sd, ".new"));
      int fd;
      file_close(cs->swaplog_fd);
- #ifdef _SQUID_OS2_
-     if (unlink(swaplog_path) < 0) {
- 	debug(50, 0) ("%s: %s\n", swaplog_path, xstrerror());
- 	fatal("storeCossDirCloseTmpSwapLog: unlink failed");
-     }
- #endif
      if (xrename(new_path, swaplog_path) < 0) {
  	fatal("storeCossDirCloseTmpSwapLog: rename failed");
      }
--- 411,416 ----
***************
*** 613,621 ****
  #ifdef _SQUID_OS2_
  	file_close(state->fd);
  	state->fd = -1;
- 	if (unlink(cur) < 0)
- 	    debug(50, 0) ("storeCossDirWriteCleanLogs: unlinkd failed: %s, %s\n",
- 		xstrerror(), cur);
  #endif
  	xrename(state->new, state->cur);
      }
--- 607,612 ----
Index: squid/src/fs/diskd/store_dir_diskd.c
diff -c squid/src/fs/diskd/store_dir_diskd.c:1.58.2.10 squid/src/fs/diskd/store_dir_diskd.c:1.58.2.11
*** squid/src/fs/diskd/store_dir_diskd.c:1.58.2.10	Sat Mar 26 15:29:25 2005
--- squid/src/fs/diskd/store_dir_diskd.c	Sat Mar 26 16:27:11 2005
***************
*** 1320,1331 ****
      char *new_path = xstrdup(storeDiskdDirSwapLogFile(sd, ".new"));
      int fd;
      file_close(diskdinfo->swaplog_fd);
- #ifdef _SQUID_OS2_
-     if (unlink(swaplog_path) < 0) {
- 	debug(50, 0) ("%s: %s\n", swaplog_path, xstrerror());
- 	fatal("storeDiskdDirCloseTmpSwapLog: unlink failed");
-     }
- #endif
      if (xrename(new_path, swaplog_path) < 0) {
  	fatal("storeDiskdDirCloseTmpSwapLog: rename failed");
      }
--- 1320,1325 ----
***************
*** 1549,1557 ****
  #ifdef _SQUID_OS2_
  	file_close(state->fd);
  	state->fd = -1;
- 	if (unlink(cur) < 0)
- 	    debug(50, 0) ("storeDirWriteCleanLogs: unlinkd failed: %s, %s\n",
- 		xstrerror(), cur);
  #endif
  	xrename(state->new, state->cur);
      }
--- 1543,1548 ----
Index: squid/src/fs/ufs/store_dir_ufs.c
diff -c squid/src/fs/ufs/store_dir_ufs.c:1.39.2.12 squid/src/fs/ufs/store_dir_ufs.c:1.39.2.13
*** squid/src/fs/ufs/store_dir_ufs.c:1.39.2.12	Sat Mar 26 15:29:26 2005
--- squid/src/fs/ufs/store_dir_ufs.c	Sat Mar 26 16:27:11 2005
***************
*** 1106,1117 ****
      char *new_path = xstrdup(storeUfsDirSwapLogFile(sd, ".new"));
      int fd;
      file_close(ufsinfo->swaplog_fd);
- #if defined (_SQUID_OS2_) || defined (_SQUID_CYGWIN_)
-     if (unlink(swaplog_path) < 0) {
- 	debug(50, 0) ("%s: %s\n", swaplog_path, xstrerror());
- 	fatal("storeUfsDirCloseTmpSwapLog: unlink failed");
-     }
- #endif
      if (xrename(new_path, swaplog_path) < 0) {
  	fatal("storeUfsDirCloseTmpSwapLog: rename failed");
      }
--- 1106,1111 ----
***************
*** 1340,1351 ****
      fd = state->fd;
      /* rename */
      if (state->fd >= 0) {
! #if defined(_SQUID_OS2_) || defined (_SQUID_CYGWIN_)
  	file_close(state->fd);
  	state->fd = -1;
- 	if (unlink(state->cur) < 0)
- 	    debug(50, 0) ("storeDirWriteCleanLogs: unlinkd failed: %s, %s\n",
- 		xstrerror(), state->cur);
  #endif
  	xrename(state->new, state->cur);
      }
--- 1334,1342 ----
      fd = state->fd;
      /* rename */
      if (state->fd >= 0) {
! #if defined(_SQUID_OS2_) || defined(_SQUID_CYGWIN_) || defined(_SQUID_MSWIN_)
  	file_close(state->fd);
  	state->fd = -1;
  #endif
  	xrename(state->new, state->cur);
      }
