--- squid-2.4.STABLE6/configure.in~	Wed Sep 12 13:46:43 2001
+++ squid-2.4.STABLE6/configure.in	Wed Apr  3 13:20:26 2002
@@ -1036,18 +1036,6 @@
 
 AC_CHECK_LIB(bsd, main)
 AC_CHECK_LIB(regex, main, [REGEXLIB="-lregex"])
-AC_CHECK_LIB(bind, gethostbyname)
-if test $ac_cv_lib_bind_gethostbyname = "no" ; then
-    case "$host" in
-	i386-*-freebsd*)
-		echo "skipping libresolv checks for $host"
-		;;
-	*)
-		AC_CHECK_LIB(resolv, inet_aton, AC_CHECK_LIB(44bsd, inet_aton))
-		AC_CHECK_LIB(resolv, main)
-		;;
-    esac
-fi
 AC_CHECK_LIB(m, main)
 
 dnl Check for libcrypt
