diff -urN squid-2.5.STABLE10.orig/helpers/external_acl/wbinfo_group/wbinfo_group.pl squid-2.5.STABLE10/helpers/external_acl/wbinfo_group/wbinfo_group.pl
--- squid-2.5.STABLE10.orig/helpers/external_acl/wbinfo_group/wbinfo_group.pl	2004-02-18 01:37:51 +0300
+++ squid-2.5.STABLE10/helpers/external_acl/wbinfo_group/wbinfo_group.pl	2005-06-24 14:25:18 +0400
@@ -17,7 +17,7 @@
 #
 
 # external_acl uses shell style lines in it's protocol
-require 'shellwords.pl';
+use Text::ParseWords qw(shellwords);
 
 # Disable output buffering
 $|=1;           
