diff -Nur squid-2.5.STABLE1.orig/configure.in squid-2.5.STABLE1/configure.in
--- squid-2.5.STABLE1.orig/configure.in	2002-11-01 11:42:29 +0500
+++ squid-2.5.STABLE1/configure.in	2002-11-01 11:43:14 +0500
@@ -2056,7 +2056,8 @@
 else
 	AC_MSG_RESULT("no")
 	echo "Will use our own inet_ntoa()."
-	LIBOBJS="$LIBOBJS inet_ntoa.o"
+	#LIBOBJS="$LIBOBJS inet_ntoa.o"
+	AC_LIBOBJ([inet_ntoa])
 #	echo "WARNING: This looks bad, and probably prevents Squid from working."
 #	echo "         If you're on IRIX and using GCC 2.8, you probably need"
 #	echo "         to use the IRIX C compiler instead."
